warehouse/localdisk.go

```diff
@@ -0,0 +1,125 @@
+package warehouse
+
+import (
+	"fmt"
+	"github.com/fullstorydev/hauser/config"
+	"github.com/nishanths/fullstory"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type LocalDisk struct {
+	conf *config.Config
+}
+
+const (
+	timestampFile string = ".sync.hs"
+)
+
+var _ Warehouse = (*LocalDisk)(nil)
+
+func NewLocalDisk(c *config.Config) *LocalDisk {
+	if c.Local.UseStartTime {
+		filename := filepath.Join(c.Local.SaveDir, timestampFile)
+		if _, err := os.Stat(filename); !os.IsNotExist(err) {
+			os.Remove(filename)
+		}
+	}
+
+	return &LocalDisk{
+		conf: c,
+	}
+}
+
+// Currently returns the names of all the fields exported by BundleFields
+func (w *LocalDisk) GetExportTableColumns() []string {
+	allFields := BundleFields()
+	fieldNames := make([]string, 0, len(allFields))
+	for name := range allFields {
+		fieldNames = append(fieldNames, name)
+	}
+	return fieldNames
+}
+
+// Reads the last sync date from file
+func (w *LocalDisk) LastSyncPoint() (time.Time, error) {
+	t := beginningOfTime
+	if w.conf.Local.UseStartTime {
+		t = w.conf.Local.StartTime
+	}
+	if _, err := os.Stat(w.conf.Local.SaveDir); os.IsNotExist(err) {
+		panic(err)
+	}
+	filename := filepath.Join(w.conf.Local.SaveDir, timestampFile)
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return t, nil
+	}
+	timebytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return t, err
+	}
+	synctime, err := time.Parse(time.RFC3339, string(timebytes))
+	return synctime, err
+}
+
+// Writes the current time to timestamp file
+func (w *LocalDisk) SaveSyncPoints(bundles ...fullstory.ExportMeta) error {
+	t := time.Now().UTC().Format(time.RFC3339)
+	if _, err := os.Stat(w.conf.Local.SaveDir); os.IsNotExist(err) {
+		panic(err)
```

Comment:
A panic is usually reserved for programmer errors, not user input errors (I'm thinking of config params in the .toml file as user input). For a non-recoverable error like a bad `SaveDir`, our strategy thus far in `hauser` is to call `log.Fatalf`, which lets you write a message stdout and then exits. Hauser users are not always 100% savvy, so even though the existing panic will probably print sufficient info to tell that the dir is missing, the panic output may bewilder Go newbies. We should wrap it with a 100% user-friendly message.

Checking the `SaveDir` is a validation that could maybe go in `NewLocalDisk`, so we fail early, and don't have to repeat the check here and in `LastSyncPoint`.

Reply:
@jameremo do you have any strong feelings about including a more human readable message for something like this where we know what the exact problem is most of the time?

Reply:
@butanian Yes, that's exactly what I was trying to say above; if we know what the exact problm is (like a missing directory), we should clearly state it, so that the user doesn't have to think/work too hard (i.e. read code or decipher a go stack trace) to fix it.

Reply:
@glarchev Could we change this `panic` call to `log.Fatalf`?

Reply:
Good catch, I actually meant to remove that check completely, since we're now checking whether the folder exists in the `NewLocalDisk` constructor.

Reply:
On the other hand, I think I'll keep it (and change it to `log.Fatalf`) -- since hauser is meant to run continuously, I suppose it's possible that someone would delete the folder while hauser is running (and `NewLocalDisk` constructor is only called once)